src/test/java/org/minions/devfund/abel/battleship/ShipTest.java

```diff
@@ -0,0 +1,300 @@
+        assertFalse(ship.shootAt(-1, 2));
+        assertFalse(ship.shootAt(-1, -1));
+        assertFalse(ship.shootAt(twentyOne, twentyOne));
+        assertFalse(ship.shootAt(0, twentyOne));
+        assertFalse(ship.shootAt(twentyOne, 0));
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testShootAtFalseZeroNineteen() {
+        boolean[] expected = {false, false, false, false, false, false, false, false};
+        ship.placeShipAt(0, 0, true, ocean);
+        final int nineteen = 19;
+        assertFalse(ship.shootAt(0, nineteen));
+        boolean[] actual = ship.getHit();
+        assertArrayEquals(expected, actual);
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testIsSunkTrue() {
+        ship.placeShipAt(0, 0, true, ocean);
+        final int seven = 7;
+        for (int i = 0; i <= seven; i++) {
+            ship.shootAt(0, i);
+        }
+        assertTrue(ship.isSunk());
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testIsSunkFalse() {
+        ship.placeShipAt(0, 0, true, ocean);
+        ship.shootAt(0, 0);
+        ship.shootAt(0, 1);
+        assertFalse(ship.isSunk());
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testToStringIsNotCompletelySunk() {
+        ship.placeShipAt(0, 0, true, ocean);
+        ship.shootAt(0, 0);
+        ship.shootAt(0, 1);
+        String expected = "S";
+        String actual = ship.toString();
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testToStringIsSunk() {
+        ship.placeShipAt(0, 0, true, ocean);
+        final int seven = 7;
+        for (int i = 0; i <= seven; i++) {
+            ship.shootAt(0, i);
+        }
+        String expected = "x";
+        String actual = ship.toString();
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testBattleshipGetShipType() {
+        final String expectedShipType = "battleship";
+        final Ship ship = new BattleShip();
+        final String actualShipType = ship.getShipType();
+        assertEquals(expectedShipType, actualShipType);
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testEmptySeaGetShipType() {
+        final String expectedShipType = "empty";
+        final Ship emptySea = new EmptySea();
+        final String actualShipType = emptySea.getShipType();
+        String expectedDot = ".";
+        String expectedDash = "-";
+        String actual = emptySea.toString();
+        assertEquals(expectedDot, actual);
+        assertEquals(expectedShipType, actualShipType);
+        assertFalse(emptySea.shootAt(0, 0));
+        assertFalse(emptySea.isSunk());
+        actual = emptySea.toString();
+        assertEquals(expectedDash, actual);
+    }
+
+    /**
+     * Test.
+     */
+    @Test
+    public void testBattleshipGetterSetter() {
+        final Ship ship = new BattleShip();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "ship" which hides the field declared at line 17. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)